docs/proposals/docs-process.md

```diff
@@ -0,0 +1,94 @@
+# Docs Process
+
+@johndmulhausen
+
+A new design for the kubernetes.io site is currently being implemented and we think it's a good time to streamline and simplify the docs creation process as we rebuild the site from scratch.
+
+When a community is empowered to be effective contributors, great things happen. That’s why I’m proposing that we move the documentation from the docs/ folder in the core Kubernetes repo in GitHub into the GitHub pages repo itself, migrate that repo into one named `kubernetes.github.io` so that it will be compatible with the GitHub Pages auto-update system, and leave all munging scripts behind.
+
+**TL;DR: Right now we do a lot of scripting and branching to create docs -- this proposal is about using Jekyll's native capability and GitHub Page's staging capability to create a much easier-to-edit set of documentation that will ship with our upcoming new site.**
```

Comment:
You've got a good starting point, since they are all munged already, but I'm dubious.  My experience was that many/most people wrote docs without looking at the rendered result.  Additionally, some of the "rules" that jekyll was demanding are non-obvious and github's own renderer does not demand, making it render correctly in a Pull Request but be wrong in jekyll.

We have gofmt and a compile check as a presubmit for a reason.  Humans are untrustworthy at syntax, and automation is needed to keep it stable.  Without those, the codebase would never compile.